File: tutorial/lect4-im/imCalibrateSignals_soln.m
run ../../utilities/initPaths.m;

buffhost='localhost';buffport=1972;
% wait for the buffer to return valid header information
hdr=[];
while ( isempty(hdr) || ~isstruct(hdr) || (hdr.nchans==0) ) % wait for the buffer to contain valid data
  try 
    hdr=buffer('get_hdr',[],buffhost,buffport); 
  catch
    hdr=[];
    fprintf('Invalid header info... waiting.\n');
  end;
  pause(1);
end;

trialDuration=3;
trlen_ms=trialDuration*1000;
dname  ='training_data';
cname  ='clsfr';

% Grab 600ms data after every stimulus.target event
[data,devents,state]=buffer_waitData(buffhost,buffport,[],'startSet',{{'stimulus.target'}},'exitSet',{'stimulus.training' 'end'},'verb',verb,'trlen_ms',trlen_ms);
mi=matchEvents(devents,'stimulus.training','end'); devents(mi)=[]; data(mi)=[]; % remove the exit event
fprintf('Saving %d epochs to : %s\n',numel(devents),dname);
save(dname,'data','devents');

File: tutorial/lect4-im/imCalibrateSignals_soln.py
#!/usr/bin/env python3
# Set up imports and paths
import sys, os
# Get the helper functions for connecting to the buffer
sigProcPath = "../signalProc"
sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)),sigProcPath))
import bufhelp 

# connect to the buffer, if no-header wait until valid connection
bufhelp.connect()

trlen_ms = 600

print("Calibration phase")
# grab data after every t:'stimulus.target' event until we get a {t:'stimulus.training' v:'end'} event 
data, events, stopevents = bufhelp.gatherdata("stimulus.target",trlen_ms,("stimulus.training","end"), milliseconds=True)
# save the calibration data
pickle.dump({"events":events,"data":data}, open("subject_data", "w"))
